.github/workflows/deploy-github-pages.yml

```diff
@@ -55,3 +55,24 @@ jobs:
           mkdir gh-pages/"$VERSION"
           mv result/html/* gh-pages/"$VERSION"
 
+      - name: Upload new artifact
+        id: upload
+        uses: actions/upload-pages-artifact@v3
+        with:
+          path: gh-pages
+
+  deploy-gh-pages:
+    needs: construct-gh-pages
+    permissions:
+      pages: write      # to deploy to Pages
+      id-token: write   # to verify the deployment originates from an appropriate source
+
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+
+    runs-on: ubuntu-latest
+    steps:
+      - name: Deploy artifact
+        id: deployment
+        uses: actions/deploy-artifact@v4
```

Comment:
Does this handle the case of versioned pages? It seems to me that this will overwrite the entire site on deployment.

Reply:
Tweaked this so that `main` gets its own folder. So now each version will go in its own directory, so adding a new version will add it to the existing set of directories in the `gh-pages` clone, so the artifact will retain all versions. This requires a setting change to serve out of `main` instead of the root, though. I've asked @kferles to do this on #111

Reply:
Alright, actually found a different workaround.